File: test_samples/vulnerable.py
# Sample vulnerable Python code for testing PatchScout

import pickle
import subprocess
import os

# Code injection vulnerabilities
def process_user_input(user_code):
    # Dangerous: eval with user input
    result = eval(user_code)
    return result

def execute_command(cmd):
    # Dangerous: exec with user input
    exec(cmd)

# SQL Injection pattern
def get_user(username):
    query = "SELECT * FROM users WHERE username = '" + username + "'"
    return query

# Command injection
def run_system_command(filename):
    os.system(f"cat {filename}")
    subprocess.call("ls " + filename, shell=True)

# Hardcoded credentials
API_KEY = "sk-abc123def456789"
PASSWORD = "SuperSecret123"

# Insecure deserialization
def load_data(serialized_data):
    data = pickle.loads(serialized_data)
    return data

# Weak cryptography
import hashlib
def hash_password(password):
    return hashlib.md5(password.encode()).hexdigest()

File: test_samples/vulnerable.php
# Sample vulnerable PHP code for testing PatchScout

<?php
// SQL Injection vulnerability
$user_id = $_GET['id'];
$query = "SELECT * FROM users WHERE id = " . $user_id;
$result = mysql_query($query);

// XSS vulnerability
echo "Welcome " . $_GET['name'];

// Command Injection
$filename = $_GET['file'];
system("cat " . $filename);

// File Inclusion
$page = $_GET['page'];
include($page . '.php');

// Hardcoded credentials
$db_password = "admin123456";
$api_key = "sk-1234567890abcdef";

// Insecure deserialization
$data = unserialize($_POST['data']);

// Weak random for session
$session_id = mt_rand(1000, 9999);
?>
